app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -469,13 +470,15 @@ private void setResultAndFinish(boolean goToJobInfo) {
 
     public void handleConnectButtonPress() {
         selectedAppIndex = -1;
-        ConnectManager.unlockConnect(this, success -> {
-            if(success) {
-                ConnectManager.goToConnectJobsList();
-                setResult(RESULT_OK);
-                finish();
-            }
-        });
+        Intent intent=new Intent(this, ConnectIdActivity.class);

```

Comment:
I don't think this is the right place to launch this activity. When the user presses the Connect button (which calls this function), they already have their ConnectID account configured so they just need to unlock and then proceed to the jobs page (as accomplished by the existing code).

The ConnectIdActivity should be launched when the option is selected from the three-dot menu. Note the same menu option exists in the CommCareSetupActivity.